src/metabase/db/metadata_queries.clj

```diff
@@ -64,3 +63,30 @@
   [field]
   (-> (field-query field (ql/aggregation {} (ql/count (ql/field-id (u/get-id field)))))
       first first int))
+
+(defn db-id
+  "Fetch the database ID of a given entity."
+  [x]
+  (db/select-one-field :db_id 'Table :id (:table_id x)))
+
+(defn field-values
+  "Return all the values of FIELD for QUERY."
+  [field query]
+  (->> (qp/process-query
+         {:type     :query
+          :database (db-id field)
+          :query    (merge {:fields       [[:field-id (:id field)]]
+                            :source-table (:table_id field)}
+                           query)})
+       :data
+       :rows
+       (map first)))
```

Comment:
yup, we should be a bit "loose" in testing the specific numbers we're calculating for an x-ray, but we should have test coverage over functions like this as well as "happy path" tests to verify that the api endpoints return results without errors for fields, tables, etc in the sample dataset.